src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -250,33 +250,36 @@ def lockout_spiders
 
   def check_user
```

Comment:
the old code did not take care of special situations like HTTP_X_USERNAME with blank string. The new code is reducing the possibilities into running of crashes.

Also it is way more readable IMHO, because the different modes are clearly seperated and not mixed.